### PR TITLE
Add HIERA compatibility

### PR DESCRIPTION
**Pull request by zyurs**
Hello,
I needed a compatibility with Hiera for this recipe. I hope you will accept it so I can use your repo instead of mine.
Thanks

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -6,6 +6,12 @@
 class git (
   $package        = $::git::params::package,
   $package_ensure = $::git::params::package_ensure,
+
+  ### START Hiera Lookups ###
+  $repo = hiera_has('git::repo', {}),
```
Comment (ekohl) [inline manifests/init.pp]: While I don't mind adding a $repos variable, I'm not a fan of hiera_has (should that be hiera_hash?) here. Personally I don't use hiera, but wouldn't a $repos variable suffice here?

**Comment by zyurs** [comment]
Yes it's hiera_hash, just a typo, I will correct it.

**Comment by zyurs** [comment]
I fixed the mistake

**Comment by ekohl** [comment]
I asked around about this because I don't know hiera well enough, but it was recommended to not add hiera_hash. We can leave in $repo with create_resources, but I'd prefer to name it $repos with a validate_hash.

**Comment by zyurs** [comment]
I understand the use of validate_hash but for hiera_hash what would you use instead ?

**Comment by ekohl** [comment]
The advice I got from #voxpupuli was not to use hiera_hash in a component module. That should be in the profile layer or in modern puppet versions in data bindings.

**Comment by mmoll** [comment]
I also have to state, I'm against bringing hiera only things into this module. I do know that paramter merging, espcially deep merging of multiple hierachies in hiera, defaults, class paramters and so on is next to impossible in Puppet, but in this case I'd say just adding the additional param and create_resources should suffice and the possible problems associated with that approach are the same like in similar places with all Puppet modules. @zyurs would that be acceptable for you?

**Comment by mmoll** [comment]
@zyurs ping?

**Comment by mmoll** [comment]
what's the status here?

**Comment by mmoll** [comment]
no reaction, closing

**Pull request closed without merging**